compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -542,22 +554,25 @@ private BSymbol getOwner(SymbolEnv symbolEnv) {
         return symbolEnv.enclPkg.symbol;
     }
 
-    private BLangExpression createClosureForDefaultValue(String closureName, String paramName,
-                                                         BLangSimpleVariable varNode) {
+    private void generateClosureForDefaultValues(String closureName, String paramName, BLangSimpleVariable varNode) {
         BSymbol owner = getOwner(env);
-        BInvokableTypeSymbol symbol = (BInvokableTypeSymbol) env.node.getBType().tsymbol;
         BLangFunction function = createFunction(closureName, varNode.pos, owner.pkgID, owner, varNode.getBType());
-        updateFunctionParams(function, symbol.params, paramName);
         BLangReturn returnStmt = ASTBuilderUtil.createReturnStmt(function.pos, (BLangBlockFunctionBody) function.body);
-        returnStmt.expr = varNode.expr;
+        returnStmt.expr = desugar.addConversionExprIfRequired(varNode.expr, function.returnTypeNode.getBType());
         BLangLambdaFunction lambdaFunction = createLambdaFunction(function);
         lambdaFunction.capturedClosureEnv = env.createClone();
         BInvokableSymbol varSymbol = createSimpleVariable(function, lambdaFunction, false);
+        BTypeSymbol symbol = env.node.getBType().tsymbol;
+        if (symbol.getKind() == SymbolKind.INVOKABLE_TYPE) {
+            updateFunctionParams(function, ((BInvokableTypeSymbol) symbol).params, paramName);
+            ((BInvokableTypeSymbol) symbol).defaultValues.put(Utils.unescapeBallerina(paramName), varSymbol);
```

Comment:
Can extract `(BInvokableTypeSymbol) symbol` out to a variable.

Reply:
Updated https://github.com/ballerina-platform/ballerina-lang/pull/37184/commits/f846176e6ff38c3e1366f21da4cfa05ea8a4fbdc